### ISSUE TITLE
0.2_4 release notes inconsistent with documentation and actual behavior of reduce functions

### ISSUE THREAD

**Issue by Enamex** [pytorch/pytorch#3995]
The [0.2](https://github.com/pytorch/pytorch/releases/tag/v0.2.0) release notes say:
> All reduce functions such as `sum` and `mean` now default to squeezing the reduced dimension. For example `torch.sum(torch.randn(10, 20))` returns a 1D Tensor.

This doesn't happen in a Python terminal with the latest pull from soumith's channel through `conda`.

The [documentation](http://pytorch.org/docs/master/torch.html?highlight=sum#torch.sum) doesn't illustrate this point either way:
```py
>>> a = torch.randn(1, 3)
>>> a

 0.6170  0.3546  0.0253
[torch.FloatTensor of size 1x3]

>>> torch.sum(a)
0.9969287421554327
```

What was the rationale behind the intended change in behavior, anyway? (It would be different from what NumPy does.) Or is this just an error in the release notes?

**Comment by soumith** [comment]
The release notes had a typo, sorry. It should've been: 
> For example torch.sum(torch.randn(10, 20), dim=0) returns a 1D Tensor.

I've fixed them now.

As you see from the documentation, there are two prototypes of `torch.sum`, one with the `dim` argument and one without. the one with the `dim` argument is a global sum of the Tensor.
